BoltApp/bolt-magento2
# EN-3825 - Assign customer an appropriate group id in backoffice order creation process

# Description
Trail Gear reported issues with customer creation during back office orders. Fixed, tested and approved in an extension made for them.The extension should be removed when this gets released and they update.

## Issue

How we did it so far is always creating a back-office order as a guest user, and never taking care of really creating a customer account and putting it into appropriate customer group. It was kind of hard-coded in the module. That's not how Magento does it and it's very strange that we missed it and no merchant pointed that out until now.

Basically, in the back-office create order flow either an existing customer is selected or a new one needs to be created in the process (Create New Customer button). No guests in Magento back-office orders. And all our back-office order customers in the create new customer flow were guests so far.

## Test cases

**Main flow - creating a new customer account with Bolt backoffice order:**

1. Go to Sales -> Orders -> Create New Order -> Create New Customer
2. Create order with Bolt keeping in mind chosen customer group and email
3. Order creation should succeed and take you to the usual order success page
4. There you need to make sure that the right column specifies the correct customer email, name and group, Also having a link saying Edit Customer is important, proving that a customer was succesfully created.
5. Optionally you can open said customer and verify its details match.

![image](https://user-images.githubusercontent.com/37150717/105889067-0e42ce80-600e-11eb-84c9-230528de48fa.png)

**A customer with the same email address already exists in an associated website.**

1. Go to Sales -> Orders -> Create New Order -> Create New Customer
2. For email address input an email address of a customer already registered with the selected website.
3. Clicking the Bolt button will display an alert saying "A customer with the same email address already exists in an associated website.".
4. Changing the customer email should create a new Bolt order and allow you to create the order.

![image](https://user-images.githubusercontent.com/37150717/105889153-2581bc00-600e-11eb-9d7f-e2960712af5e.png)

Fixes: https://boltpay.atlassian.net/browse/EN-3825, https://boltpay.atlassian.net/browse/ON-438

#changelog EN-3825 - Assign customer an appropriate group id in backoffice order creation …

# Type of change



# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [x] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.
